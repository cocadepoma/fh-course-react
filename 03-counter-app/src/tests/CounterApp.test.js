import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import CounterApp from "../CounterApp";

describe("pruebas en <CounterApp/>", () => {
    let wrapper = shallow(<CounterApp />);

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test("probar que el valor por defecto de 'value' es 10", () => {
        expect(wrapper).toMatchSnapshot();
        const numero = parseInt(wrapper.find("h2").text().trim());
        expect(numero).toBe(10);
    });

    test("probar que el 'value' de 100 pasado como prop, corresponde", () => {
        const numero = 100;

        const wrapper = shallow(<CounterApp value={numero} />);
        const numeroDom = parseInt(wrapper.find("h2").text().trim());

        expect(numeroDom).toBe(numero);
    });

    test("debe de incrementar el contador en 1 por cada click", () => {
        // const btn1 = wrapper.find("button").at(0);
        // console.log(btn1.html());
        const numeroDom = parseInt(wrapper.find("h2").text().trim());
        wrapper.find("button").at(0).simulate("click");
        const numeroDom2 = parseInt(wrapper.find("h2").text().trim());

        expect(numeroDom2).toBe(numeroDom + 1);
    });
    test("debe de decrementar el contador en 1 por cada click", () => {
        const numeroDom = parseInt(wrapper.find("h2").text().trim());
        wrapper.find("button").at(2).simulate("click");
        const numeroDom2 = parseInt(wrapper.find("h2").text().trim());

        expect(numeroDom2).toBe(numeroDom - 1);
    });

    test("debe de resetear el 'value' al valor por defecto pasado como prop", () => {
        const valor = 105;
        const wrapper = shallow(<CounterApp value={valor} />);
        wrapper.find("button").at(0).simulate("click");
        wrapper.find("button").at(0).simulate("click");
        wrapper.find("button").at(1).simulate("click");
        const numeroDom = parseInt(wrapper.find("h2").text().trim());

        expect(numeroDom).toBe(valor);
    });
});
