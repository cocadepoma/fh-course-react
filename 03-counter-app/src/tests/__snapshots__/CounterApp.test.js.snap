// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp/> probar que el valor por defecto de 'value' es 10 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
